src/contracts/ParaSwapLiquiditySwapAdapter.sol

```diff
@@ -64,23 +64,21 @@ abstract contract ParaSwapLiquiditySwapAdapter is
   /// @inheritdoc IParaSwapLiquiditySwapAdapter
   function swapLiquidity(
     LiquiditySwapParams memory liquiditySwapParams,
-    FlashParams memory flashParams,
     PermitInput memory collateralATokenPermit
   ) external nonReentrant {
     // Offset in August calldata if wanting to swap all balance, otherwise 0
     if (liquiditySwapParams.offset != 0) {
       (, , address aToken) = _getReserveData(liquiditySwapParams.collateralAsset);
-      uint256 balance = IERC20(aToken).balanceOf(liquiditySwapParams.user);
-      require(balance <= liquiditySwapParams.collateralAmountToSwap, 'INSUFFICIENT_AMOUNT_TO_SWAP');
```

Comment:
this is a sanity check we must keep.
"If the user wants to swap all, the amount used for the API call and passed as argument should be more than the full balance"